packages/flutter/lib/src/material/carousel.dart

```diff
@@ -1136,6 +1197,34 @@ class _RenderSliverWeightedCarousel extends RenderSliverFixedExtentBoxAdaptor {
     double itemExtent,
   ) {
     final int? childCount = childManager.estimatedChildCount;
+
+    // For infinite scrolling, calculate how many items fit in the viewport
+    if (infinite && childCount == null) {
+      double visibleItemsTotalExtent = _distanceToLeadingEdge;
+      int index = _firstVisibleItemIndex + 1;
+      // Calculate upper bound based on viewport extent and minimum possible item extent
+      // In worst case, all items would be at minimum extent (minChildExtent)
+      // Using weights.length as a multiplier provides a reasonable buffer
+      final double safeMinExtent = math.max(minChildExtent, 1.0);
+      final int estimatedUpperBound =
+          _firstVisibleItemIndex +
+          clampDouble(
+            ((constraints.viewportMainAxisExtent / safeMinExtent).ceil() * weights.length)
```

Comment:
`(constraints.viewportMainAxisExtent / safeMinExtent).ceil()` is how many items with minimum extent in the viewport, but why do we need to multiply it by weights length? Feels the result doesn't make sense.

seems `(constraints.viewportMainAxisExtent / safeMinExtent).ceil()` is safe enough?